crates/biome_js_analyze/src/lint/nursery/use_readonly_class_properties.rs

```diff
@@ -0,0 +1,449 @@
+}
+
+fn is_default<T: Default + Eq>(value: &T) -> bool {
+    value == &T::default()
+}
+
+type TextRangeWithText = (TextRange, Text, SyntaxNode<JsLanguage>);
+
+fn filter_mutable_class_property(
+    member: &AnyJsClassMember,
+    private_only: bool,
+) -> Option<TextRangeWithText> {
+    let property_class_member = member.as_js_property_class_member()?;
+
+    if property_class_member
+        .modifiers()
+        .iter()
+        .any(|x| TsReadonlyModifier::cast(x.into_syntax()).is_some())
+    {
+        return None;
+    }
+
+    let name = property_class_member.name().ok()?;
+    let text_range_and_name = Some((
+        name.range(),
+        name.to_trimmed_text(),
+        property_class_member.syntax().clone(),
+    ));
+
+    if !private_only {
+        return text_range_and_name;
+    }
+
+    let is_es_private = matches!(
+        member.name().ok()??,
+        AnyJsClassMemberName::JsPrivateClassMemberName(_)
+    );
+    let is_ts_private = property_class_member.modifiers().iter().any(|x| {
+        TsAccessibilityModifier::cast(x.into_syntax()).is_some_and(|modifier| modifier.is_private())
+    });
+
+    if is_es_private || is_ts_private {
+        return text_range_and_name;
+    }
+    None
+}
+
+fn filter_mutable_constructor_parameters(
+    class_declaration: &JsClassDeclaration,
+    private_only: bool,
+) -> Option<Vec<TextRangeWithText>> {
+    let constructor = class_declaration
+        .members()
+        .iter()
+        .find_map(|member| match member {
+            AnyJsClassMember::JsConstructorClassMember(member) => Some(member),
+            _ => None,
+        })?;
+
+    let constructor_params = constructor.parameters().ok()?;
+    let params: Vec<TextRangeWithText> = constructor_params
+        .parameters()
+        .iter()
+        .filter_map(|param| match param.ok()? {
+            AnyJsConstructorParameter::TsPropertyParameter(ts_property) => Some(ts_property),
+            _ => None,
+        })
+        .filter(|param| {
+            let is_mutable = param
+                .modifiers()
+                .iter()
+                .find_map(|x| TsReadonlyModifier::cast(x.into_syntax()))
+                .is_none();
+
+            let is_private = param.modifiers().iter().any(|x| {
+                x.as_ts_accessibility_modifier()
+                    .is_some_and(|modifier| modifier.is_private())
+            });
+
+            is_mutable && (!private_only || is_private)
+        })
+        .filter_map(|ts_parameter| {
+            if let Ok(binding) = ts_parameter
+                .formal_parameter()
+                .ok()?
+                .as_js_formal_parameter()?
+                .binding()
+            {
+                let syntax = binding.syntax();
+                let text_range = syntax.text_trimmed_range();
+                let text = syntax.text_trimmed().into_text();
+
+                return Some((text_range, text, ts_parameter.syntax().clone()));
+            }
+            None
```

Comment:
```suggestion
        .filter_map(|ts_parameter| {
            let binding = ts_parameter
                .formal_parameter()
                .ok()?
                .as_js_formal_parameter()?
                .binding().ok()?;

                let syntax = binding.syntax();
                let text_range = syntax.text_trimmed_range();
                let text = syntax.text_trimmed().into_text();

                return Some((text_range, text, ts_parameter.syntax().clone()));
```

This doesn't require a `if let OK` anymore